sdk/core/core/src/az_log_private.h

```diff
@@ -19,6 +19,8 @@ void _az_log_http_response(
     uint64_t const duration_msec,
     az_http_request_builder const * const hrb);
 
+void _az_log_http_retry(int16_t const attempt, int32_t const delay_msec);
```

Comment:
I don't think we need this in any header file if it's just called from the retry policy. Maybe we want to move the retry policy and all its supporting functions to its own .c file. Then, only the retry policy function and its options struct need to be in a header file.

Reply:
I think it is more convenient to keep logging-related private functions in a separate file so that they could reuse/share code/formatting parts if necessary.

Reply:
No, this is not the right thing to do. It is better to keep all retry-related logic together.
Logging is a cross-cutting concern; it may appear anywhere and everywhere in the code base. We should NOT have to edit a logging .c file to add a log entry to a new classification somewhere else in the code base.